### ISSUE TITLE
scroll container max-height is calculated (does not render entire list)

### ISSUE THREAD

**Issue by deepflame** [html-next/vertical-collection#101]
Hi everyone,

thanks for the great addon!
Today I just ran into a problem.

The list is not completely rendered and just fills up when I start to scroll it.
Guess the reason is that the surrounding scroll container has a calculated height like this:

```css
  max-height: calc(100vh - 170px);
```

Any idea how to redraw the list properly?

Thanks
Andreas

**Comment by pzuraq** [comment]
Hey @deepflame,

We don't currently handle max-height properties that are non-pixel based, you can see the relevant code here: https://github.com/html-next/vertical-collection/blob/master/addon/-private/data-view/radar/radar.js#L182

I think supporting non-pixel based values would be fine, it's just tricky to implement.

**Comment by deepflame** [comment]
Hi @pzuraq ,

thanks a lot for the quick reply!

Hmm, it seems that the `scrollHeight` values I get here are both 0 ( https://github.com/html-next/vertical-collection/blob/master/addon/-private/data-view/radar/radar.js#L186 ).
It should take whatever is bigger, right?

Seems to me this is a timing issue, no? Guess the height of the container is not known at the time its bounds are known ( https://github.com/html-next/vertical-collection/blob/master/addon/-private/data-view/radar/radar.js#L180 )

Is there any way to force a redraw? The list renders correctly when I resize the window or scroll it...

**Comment by pzuraq** [comment]
I'm thinking that it's because you're using `max-height` instead of `height`. The container will conform to its content, and because there are no items rendered at all on initial render it will be 0. If your `max-height` value were a static pixel number we could just use that instead, which is what happens now, but because it's based on non-pixel units we need to add functionality to convert those units to pixels. In this case, we'd have to get the bounding rect of the viewport, take the height, and subtract 170px.

Where this gets tricky is with other CSS expressions - what about percentages? Those are relative to certain parents, not necessarily the viewport. And interpreting the `calc` property to get an absolute value will also be hard.

My recommendation for the moment is to try to give your scroll container an actual height. Even if you're using CSS `calc`, because then the scrollContainer will be measured directly and everything will work. Alternatively, make your `max-height` property an absolute pixel value.

**Comment by deepflame** [comment]
Hi @pzuraq ,

I did as you suggested and put `height: 100vh`. This is working perfectly. So the initial render takes up the whole space (but still honoring the max-height value).

I also learned that `element.style.maxHeight` just works on inline styles. So no wonder I just got an empty string.

Sure, putting the max-height inline and parsing its complex CSS would just be too much. Setting just the height is so much easier :)

Thanks for your time and effort
Andreas
